test/Extension.tests/JobhostEndToEnd.cs

```diff
@@ -183,5 +266,62 @@ public void TestCloudEventToJObject([EventGridTrigger] JObject value)
                 _functionOut = (string)value["eventType"];
             }
         }
+
+        public class OutputBindingParams
+        {
+            [return: EventGrid(TopicEndpointUri = "eventgridUri", SasKey = "eventgridKey")]
+            public EventGridEvent TestOutputTypes(
+                [EventGrid(TopicEndpointUri = "eventgridUri", SasKey = "eventgridKey")] out EventGridEvent single,
+                [EventGrid(TopicEndpointUri = "eventgridUri", SasKey = "eventgridKey")] out EventGridEvent[] array,
+                [EventGrid(TopicEndpointUri = "eventgridUri", SasKey = "eventgridKey")] ICollector<EventGridEvent> collector,
+                [EventGrid(TopicEndpointUri = "eventgridUri", SasKey = "eventgridKey")] IAsyncCollector<EventGridEvent> asyncCollector)
+            {
+                // does not actually send, custruct simplest event possible
+                single = new EventGridEvent()
+                {
+                    EventType = "singleEvent",
+                    Data = 0
+                };
+
+                array = new EventGridEvent[5];
+                for (int i = 0; i < 5; i++)
+                {
+                    array[i] = new EventGridEvent()
+                    {
+                        EventType = "arrayEvent",
+                        Data = i
+                    };
+                }
+
+                for (int i = 0; i < 4; i++)
+                {
+                    collector.Add(new EventGridEvent()
+                    {
+                        EventType = "collectorEvent",
+                        Data = i
+                    });
+                }
+
+                for (int i = 0; i < 7; i++)
+                {
+                    asyncCollector.AddAsync(new EventGridEvent()
+                    {
+                        EventType = "asyncCollectorEvent",
+                        Data = i
+                    }).Wait();
```

Comment:
Rather than sprinkle these Wait() calls in here, can you just make this function async (`public async Task<EventGridEvent> TestOutputTypes`)?

Reply:
no because `async` methods can't have `ref` or `out` params, and this one does.

Reply:
No opinion. I didn't write the original implementation of this (not sure why Blame shows differently) so I'm fine leaving it or refactoring.

Reply:
now IAsyncCollector is in its own async method